bindings/kafka/metadata.yaml

```diff
@@ -121,6 +121,50 @@ authenticationProfiles:
         example: |
           {"cluster":"kafka","poolid":"kafkapool"}
         type: string
+      - name: oidcClientAuthMethod
+        type: string
+        required: false
+        default: '"client_secret"'
+        description: |
+          Client authentication method at the token endpoint.
+          The "client_secret" method (default) uses the client_id and client_secret to authenticate.
+          The "client_jwt" method uses a JWT client assertion to authenticate.
```

Comment:
would it make sense to instead of supporting a field for oidcClientAuthMethod to instead make the current oidc auth profile to be the client_secret approach, and then to add a secondary higher level auth profile for OIDC JWT Client assertion? That way then the metadata fields are cleanly separated according to the actual auth profile a user is going for so it is a cleaner DX? This would remove the fields such as `oidcClientAssertionCert` to not have conditionals on if it is required or not.